File: Leetcode/Leetcode-DSA-course/Hashing/2053-check-if-all-characters-have-equal-number-of-occurrences/check-if-all-characters-have-equal-number-of-occurrences.ts
function areOccurrencesEqual(s: string): boolean {
    const counts = new Map();
    for (let ch of s)
        counts.set(ch, (counts.get(ch) || 0) + 1)

    const frequencies = new Set(counts.values())
    return frequencies.size == 1;
};

File: Leetcode/Leetcode-DSA-course/Hashing/2053-check-if-all-characters-have-equal-number-of-occurrences/check-if-all-characters-have-equal-number-of-occurrences.java
class Solution {
    public boolean areOccurrencesEqual(String s) {
        Map<Character, Integer> counts = new HashMap<>();
        for (char ch : s.toCharArray())
            counts.put(ch, counts.getOrDefault(ch, 0) + 1);
        Set<Integer> frequencies = new HashSet<>(counts.values());
        return frequencies.size() == 1;
    }
}